layer/scripts/gen_all_ext_profile.py

```diff
@@ -0,0 +1,530 @@
+        self.i += 1
+        gen = ''
+        first = True
+        for i in range(num):
+            if first:
+                first = False
+            else:
+                gen += ', '
+            index = int(self.pseudo_random(i) * 1000) % len(registry.format_features)
+            gen += '\"'
+            gen += registry.format_features[index]
+            gen += '\"'
+
+        if num > 0:
+            gen += ' '
+
+        return gen
+
+    def gen_formats(self, registry):
+        gen = ''
+        first = True
+        for format in registry.formats:
+            if (first):
+                first = False
+            else:
+                gen += ','
+            gen += '\n                \"'
+            gen += format
+            gen += '\": {\n'
+            gen += '                    "VkFormatProperties": {\n'
+            gen += '                        "linearTilingFeatures": [ '
+            gen += self.gen_random_format_features(registry)
+            gen += '],\n'
+            gen += '                        "optimalTilingFeatures": [ '
+            gen += self.gen_random_format_features(registry)
+            gen += '],\n'
+            gen += '                        "bufferFeatures": [ '
+            gen += self.gen_random_format_features(registry)
+            gen += ']\n'
+            gen += '                    }\n'
+            gen += '                }'
+        return gen
+
+    def gen_privateImpl(self, registry):
+        gen = '{\n'
+        gen += '    "$schema": "https://schema.khronos.org/vulkan/profiles-1.3.204.json#",\n'
+        gen += '    "capabilities": {\n'
+        gen += '        "baseline": {\n'
+        gen += '            "extensions": {'
+        gen += self.gen_extensions(registry.instance_extensions, True)
+        gen += self.gen_extensions(registry.device_extensions, not bool(registry.instance_extensions))
+        gen += '\n'
+        gen += '            },\n'
+        gen += '            "features": {'
+        gen += self.gen_features(registry.instance_extensions, registry, True)
+        gen += self.gen_features(registry.device_extensions, registry, not bool(registry.instance_extensions))
+        gen += '\n            },\n'
+        gen += '            "properties": {'
+        gen += self.gen_properties(registry.instance_extensions, True, registry)
+        gen += self.gen_properties(registry.device_extensions, not bool(registry.instance_extensions), registry)
+        gen += '\n            },\n'
+        gen += '            "formats": {'
+        gen += self.gen_formats(registry)
+        gen += '\n            }\n'
+        gen += '        }\n'
+        gen += '    },'
+        gen += PROFILES
+        gen += '}\n'
+        return gen
+
+class VulkanExtension():
+    def __init__(self, spec_version, feature, property):
+        self.spec_version = spec_version
+        self.feature = feature
+        self.property = property
+
+class VulkanRegistry():
```

Comment:
It seems like you reused part of the infrastructure in `genvp.py`. I think maybe a better idea would be to factor out the reusable components (e.g. `VulkanRegistry` and related classes), unless you intentionally departed from some of the data layouts used there.